WIP: Added GUI elements for tagging, added REST API

devos50 opened the pull request.

devos50 committed:
Made horizontal margin of DialogContainer flexible

devos50 committed:
Added GUI setting to disable tags

devos50 committed:
Fixed fetching tag GUI settings

drew2a commented on src/tribler-core/tribler_core/components/tag/db/tag_db.py:
```diff
@@ -17,7 +18,15 @@ class TagDatabase:
+            create_db = not filename.is_file()
+            db_filename = str(filename)
+
+        self.instance.bind('sqlite', db_filename, create_db=create_db)
```
@kozlovsky is it necessary to explicitly set `create_db`?

devos50 replied:
Note that I copied this logic from the `MetadataStore` logic. I needed an actual database to work with 👍

devos50 committed:
Added GUI prototype

devos50 committed:
Written GUI tests

drew2a committed:
Add TagDatabase

drew2a committed:
Add Tag REST

drew2a committed:
Add TagCrypto

drew2a committed:
Add TagRequestController

drew2a committed:
Add TagValidator

drew2a committed:
Add TagCommunity

drew2a committed:
Add TagComponent

drew2a committed:
Fixes from PR

devos50 committed:
Filtering tags in the REST API

drew2a committed:
Simplify test_get_my_channel_tags_xxx

drew2a committed:
Add TagDb.get_tags_operations_for_gossip()

drew2a committed:
Change timer to counter

devos50 committed:
Converting tags to lower-case in REST API

devos50 committed:
Ensure that random tags during tests are lowercase

drew2a committed:
Add lower-case validation to TagValidator

drew2a committed:
Extend Tests

drew2a committed:
Update IPV8 pointer to dataclass wrapper feature

drew2a committed:
Apply ipv8 dataclass

drew2a committed:
Revert Crypto to a more secure format

drew2a committed:
Simplify validator

devos50 committed:
Closing edit tags dialog on ESC press

devos50 committed:
Modified shutdown procedure in GUI tests

In this PR, I slightly modify the shutdown procedure in the GUI tests. I dived a bit into the shutdown procedure when I triggered a segmentation fault in Qt when using dynamic row heights for tags (by overriding the sizeHint method). This segmentation fault would trigger in ~10% of the test runs, and by investigating the (fortunately, symbolicated!) stack trace, it seems to happen when Qt tries to do a final render operation when closing the Qt application. While a more effective solution is probably out of reach for now, it seems that we can workaround this by more gracefully shutting down the components in the GUI tests. I also believe this is a more reliable approach to unload the GUI tests.

Specifically, this PR ensures that we move to the 'shutting down' screen in the GUI when the GUI tests are done (and a screenshot is also made). Then, we shutdown the core. Only when the core process is finished, we close the Qt application.

Also, by making the `window` fixture dependent on the `tribler_api` fixture, I was able to remove the `tribler_api` fixture declaration from the test methods.
